agent/taskresource/volume/dockervolume.go

```diff
@@ -15,22 +15,42 @@ package volume
 
 import (
 	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
 	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
 
 	"encoding/json"
 	"sync"
 	"time"
 )
 
-// VolumeResource represents volume resource
+const (
+	// TaskScope indicates that the volume is created and deleted with task
+	TaskScope = "task"
```

Comment:
It seems like both `TaskScope ` and `SharedScope` should be enums. wdyt?

Reply:
yes, I agree `enums` makes it looks better, but using constants here makes it simple to avoid conversion between string and enums, as this is what backend send to us. I think they works the same as enums is implemented based on constants, so I will keep this as it is unless you strongly want to change this.